src/bin/am/commands/start.rs

```diff
@@ -639,10 +634,15 @@ async fn start_prometheus(
 async fn start_pushgateway(
     pushgateway_path: &Path,
     ephemeral: bool,
-    external_url: &SocketAddr,
+    rx: Receiver<Option<SocketAddr>>,
 ) -> Result<()> {
     let work_dir = AutoCleanupDir::new("pushgateway", ephemeral)?;
 
+    let external_url = rx.borrow().map_or_else(
+        || "localhost:6789".to_string(),
+        |address| address.to_string(),
+    );
```

Comment:
Maybe we can use a `wait_for` here. To make sure that the value is not `None`.

Some thing like (haven't tested it):

```
// unwrap should be safe, since we checked for `is_some`
rx.wait_for(|valu| val.is_some()).map(|val| val.to_string()).unwrap()
```